preprocess_web/code/ravens_metadata_apps/preprocess_jobs/views.py

```diff
@@ -125,6 +125,38 @@ def view_api_retrieve_rows(request):
         return JobUtil.retrieve_rows_csv(request, job, **frm.cleaned_data)
 
 
+@csrf_exempt
+def variable_display_endpoint(request):
+    """ API endpoint to get JSON request for updating Preprocess file"""
+    if request.method != 'POST':
+        user_msg = dict(success=False,
+                            message='Please use a POST to access this endpoint')
+        return JsonResponse(user_msg)
+    # get json file from POST using view_helper
+    update_json = get_request_body_as_json(request)[1]
```

Comment:
Check for an error here and return any error messages in a JsonResponse.  e.g. don't just pick off the 2nd value [1]
- the method will find error such as badly formatted JSON, no JSON, etc

Reply:
example usage:
```python
    success, json_data_or_err = get_request_body_as_json(self.request_obj)
        if not success:
            user_msg = dict(success=False,
                                        message=json_data_or_err)
           return JsonResponse(user_msg)